File: task-03/C++/3/Sub-Task3.cpp
#include <iostream>
#include <iomanip>

using namespace std;

void printDiamond(int n) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < (n - i - 1); j++) {
            cout << " ";
        }
        for (int j = 0; j < (2 * i + 1); j++) {
            cout << "*";
        }
        cout << endl;
    }

    for (int i = n - 2; i >= 0; i--) {
        for (int j = 0; j < (n - i - 1); j++) {
            cout << " ";
        }
        for (int j = 0; j < (2 * i + 1); j++) {
            cout << "*";
        }
        cout << endl;
    }
}

int main() {
    int n;
    cout << "Enter a number: ";
    cin >> n;

    printDiamond(n);

    return 0;
}

File: task-03/Go/3/Sub-Task3.go
package main

import (
	"fmt"
)

func printDiamond(n int) {
	for i := 0; i < n; i++ {
		for j := 0; j < (n - i - 1); j++ {
			fmt.Print(" ")
		}
		for j := 0; j < (2*i + 1); j++ {
			fmt.Print("*")
		}
		fmt.Println()
	}

	for i := n - 2; i >= 0; i-- {
		for j := 0; j < (n - i - 1); j++ {
			fmt.Print(" ")
		}
		for j := 0; j < (2*i + 1); j++ {
			fmt.Print("*")
		}
		fmt.Println()
	}
}

func main() {
	var n int
	fmt.Print("Enter a number: ")
	fmt.Scan(&n)

	printDiamond(n)
}
